bot.py

```diff
@@ -457,13 +458,35 @@ async def countdown(self, message, command_content):
         except ValueError:
             await message.channel.send(f":robot: {command_content} is not an integer.")
 
+    async def interrupt_play(self, message, source):
+        """
+        Pause currently playing audio c (if any), play source s, then resume c
+        """
+        voice_client = await self.create_or_get_voice_client(message)
+        if not voice_client:
+            return
+
+        def after_interrupt(_error):
```

Comment:
How about encapsulating this instead in a class with a __call__ method?

```
class AfterInterrput:
   def __init__(self, source, after_callback):
       self.source = source
       self.after_callback = after_callback

   def __call__(self, *args):
        if self.source:
           self.voice_client.play(self.source, after = self.after_callback)

...
...
voice_client.play(source, after = AfterInterrput(voice_client.source, self.after_callback))
```

Reply:
Just tried your suggestion, doesn't work. Only the latest `after` callback is called.

Reply:
What is the benefit to using a class over a closure?

Reply:
Using a class makes the variable dependencies explicit. For example, current_source is used in the after_interrupt function but defined after 'after_interrupt' in the code, which is confusing to me.

Reply:
Done